src/Microsoft.Identity.Web/ClaimsPrincipalExtensions.cs

```diff
@@ -16,7 +16,7 @@ public static class ClaimsPrincipalExtensions
         /// </summary>
         /// <param name="claimsPrincipal">Claims principal.</param>
         /// <returns>A string corresponding to an account identifier as defined in <see cref="Microsoft.Identity.Client.AccountId.Identifier"/>.</returns>
-        public static string GetMsalAccountId(this ClaimsPrincipal claimsPrincipal)
+        public static string? GetMsalAccountId(this ClaimsPrincipal claimsPrincipal)
```

Comment:
If a consuming application also enforces the "nullable reference" feature, then they will get **warnings** when they try to pass in a NULL, for example a null ClaimsPrincinpal here. 

If the consuming application does not enforce this feature, they will not get warnings.

Reply:
How do we decide if we should change the parameter to nullable or keep it non-nullable and throw the ArgumentNullException? 

In this case, a null claimsPrincipal can return a null ID value and non-null claimsPrincipal returns non-null ID value.

Reply:
At line 23 we throw an `ArgumentNullEx` if the claimsPrincipal is null, we don't return null.